bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java

```diff
@@ -46,23 +65,59 @@
     private HashMap<String, Integer> typeTable = new HashMap<>();
     protected HashMap<String, String> stateNames = new HashMap<>();
 
+    protected String url = "";
+
+    private Map<String, Unit<?>> units = new HashMap<>();
+
     public SomfyTahomaBaseThingHandler(Thing thing) {
         super(thing);
+        // Define default units
+        units.put("Number:Temperature", SIUnits.CELSIUS);
+        units.put("Number:Energy", Units.WATT_HOUR);
+        units.put("Number:Illuminance", Units.LUX);
+        units.put("Number:Dimensionless", Units.PERCENT);
     }
 
     public HashMap<String, String> getStateNames() {
         return stateNames;
     }
 
-    protected String url = "";
-
     @Override
     public void initialize() {
-        url = getURL();
-        if (getThing().getProperties().containsKey(RSSI_LEVEL_STATE)) {
-            createRSSIChannel();
+        logger.debug("initializing thing handler for thing {}", getThing().getUID());
+        Bridge bridge = getBridge();
+        initializeThing(bridge != null ? bridge.getStatus() : null);
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("bridgeStatusChanged {} for thing {}", bridgeStatusInfo, getThing().getUID());
+        initializeThing(bridgeStatusInfo.getStatus());
+    }
+
+    public void initializeThing(@Nullable ThingStatus bridgeStatus) {
+        SomfyTahomaBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null && bridgeStatus != null) {
+            url = getURL();
+            if (getThing().getProperties().containsKey(RSSI_LEVEL_STATE)) {
+                createRSSIChannel();
+            }
+            if (bridgeStatus == ThingStatus.ONLINE) {
+                SomfyTahomaDevice device = bridgeHandler.getCachedDevice(url);
+                if (device != null) {
+                    updateUnits(device.getAttributes());
+                    List<SomfyTahomaState> states = device.getStates();
+                    updateThingStatus(states);
+                    updateThingChannels(states);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, UNAVAILABLE);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
```

Comment:
Actually, these states are set by the framework, if the bridge is offline. Is there a reason why you set these explicitly?

Reply:
The default behaviour of `bridgeStatusChanged` is to set the thing ONLINE when the bridge becomes ONLINE. That's all. In our case, we need to run additional tasks when it occurs like setting channels. That is the reason why `bridgeStatusChanged` is ovverriden. Because it is overriden, I have to handle all cases, including the case the bridge becomes OFFLINE.

Reply:
OK